### PR TITLE
[8.0][event_track_generate] Quick event tracks generator

### PR DESCRIPTION
**Pull request by yajo**
Aiming for https://github.com/OCA/l10n-spain/issues/174, this module generates event tracks automatically.

### CODE REVIEW

File: event_track_generate/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# © 2015 Grupo ESOC Ingeniería de Servicios, S.L.U.
+
+{
+    "name": "Generate Event Tracks",
+    "version": "8.0.2.0.0",
+    "category": "Project",
+    "author": "Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "website": "http://www.grupoesoc.es",
+    "summary": "Insert automatically event tracks",
```
Comment (pedrobaeza) [inline event_track_generate/__openerp__.py]: s/Insert automatically event tracks/Generate a bunch of event tracks at once

File: event_track_generate/i18n/es.po [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+#: view:event_track_generate.generator:event_track_generate.generator_view_form
+msgid "Weekdays"
+msgstr "Días de la semana"
+
+#. module: event_track_generate
+#: view:event_track_generate.generator:event_track_generate.generator_view_form
+msgid "or"
+msgstr "o"
+
+#~ msgid "Dates"
```
Comment (pedrobaeza) [inline event_track_generate/i18n/es.po]: Remove this

File: event_track_generate/models.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        This is the main method of this class, triggered by the UI.
+        """
+        # You need at least one weekday
+        weekdays = self.weekdays()[0]
+        if True not in weekdays:
+            raise exceptions.NoWeekdaysError()
+
+        # Delete existing
+        if self.delete_existing_tracks:
+            self.event_id.track_ids.exists().unlink()
```
Comment (pedrobaeza) [inline event_track_generate/models.py]: Remove `exists()`. The unlink is done the same way, independently if there are records or not

File: event_track_generate/models.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+    @api.one
+    def action_generate(self):
+        """Generate event tracks according to received data.
+
+        This is the main method of this class, triggered by the UI.
+        """
+        # You need at least one weekday
+        weekdays = self.weekdays()[0]
+        if True not in weekdays:
```
Comment (pedrobaeza) [inline event_track_generate/models.py]: `if not any(weekdays):`

**Comment by rafaelbn** [comment]
Rebuilding runbot for testing. Please @Yajo could you review las comments of @pedrobaeza ? Thanks!

File: event_track_generate/views/event.xml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                    <field name="adjust_start_time"/>
+                    <field name="adjust_end_time"/>
+                </group>
+            </group>
+            <footer>
+                <button name="action_generate"
+                        type="object"
+                        string="Generate tracks"
+                        class="oe_highlight"/>
+                or
```
Comment (yajo) [inline event_track_generate/views/event.xml]: @pedrobaeza I cannot delete the _or_; I need it here.

Comment (pedrobaeza) [inline event_track_generate/views/event.xml]: I'm not talking about the or, but the last line in translation file that it's commented

Comment (yajo) [inline event_track_generate/views/event.xml]: Ah OK, done then.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by rafaelbn** [comment]
Thanks! Very useful. I cannot test it in runbot. Not sure why...

**Comment by rafaelbn** [comment]
Tested in local :+1:

**Pull request merged**